src/config/mod.rs

```diff
@@ -614,6 +614,10 @@ macro_rules! build_cf_opt {
             } else {
                 warn!("compaction guard is disabled due to region info provider not available")
             }
+        } else {
+            // If compaction guard is disabled, we need to set multiplier to reduce file
+            // count.
+            cf_opts.set_target_file_size_multiplier(2);
```

Comment:
This should be a v2 optimization instead of guard related. The multiplier has been 1 when guard isn't implemented before. We don't know how changing it to 2 will affect older clusters.